### PR TITLE
teamchains: hidden chain tests, first batch

### PR DESCRIPTION
**Pull request by maxtaco**
* introduce determinism into all teamchains
* allow to skip tests that aren't currently working
* cover about ~13 good teamchain failures with hidden chains
* simple_hidden, which is the happy path, to make sure it all works when correct
* part 1, expect part 2 in another PR
* Works with v1.0.6 of https://github.com/keybase/node-forge-sigchain, now live

### CODE REVIEW

**Comment by heronhaye** [comment]
I can ignore all the json files?

File: teamchains/inputs/c.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+# Example Usage:
+#   ./compile.sh inputs/*.iced
```
Comment (heronhaye) [inline teamchains/inputs/c.sh]: rename file?

File: teamchains/inputs/c.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+for i in $*; do
+    input=$i
+    output=`basename -s .iced $i`.json
```
Comment (heronhaye) [inline teamchains/inputs/c.sh]:
```suggestion
    output=$(basename -s .iced "$i").json
```

File: teamchains/inputs/hidden_bad_link_type.iced [hunk 3]
```diff
@@ -0,0 +1,30 @@
+description: "Simplest test"
```
Comment (heronhaye) [inline teamchains/inputs/hidden_bad_link_type.iced]: might be confusing to have same description for all the files if it's displayed in errors somewhere

**Comment by maxtaco** [comment]
yes! they are auto-generated from the .iced files

File: teamchains/inputs/hidden_bad_parent.iced [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      { type : "rotate_key" },
+    ]
+  }
+}
+
+sessions: [
+  { loads : [{
+    error : true
+    error_type : "ParseError"
+    error_substr : "unknown link type 1000"
```
Comment (heronhaye) [inline teamchains/inputs/hidden_bad_parent.iced]: ?

File: teamchains/inputs/hidden_repeated_ptk_type.iced [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+teams: {
+  cabal : {
+    links : [
+      { type: "root", members: { owner: ["herb"] } }
+      { type : "rotate_key" }
+      {
+        type : "rotate_key_hidden",
+        corruptors : {
+          sig_arg : (arg) ->
+            arg.per_team_keys.push arg.per_team_keys[0]
```
Comment (heronhaye) [inline teamchains/inputs/hidden_repeated_ptk_type.iced:16]: test unknown ptk type as well?

File: teamchains/inputs/hidden_bad_parent.iced [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+teams: {
+  cabal : {
+    links : [
+      { type: "root", members: { owner: ["herb"] } },
+      { type : "rotate_key" },
+      {
+        type : "rotate_key_hidden"
+        corruptors :
+          sig_arg : (arg) ->
+            arg.parent_chain_tail.hash[0] ^= 0x01
```
Comment (heronhaye) [inline teamchains/inputs/hidden_bad_parent.iced:16]: also, would be good to check for seqno and seqtype mismatch as well

File: teamchains/inputs/hidden_bad_inner_hash.iced [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+teams: {
+  cabal : {
+    links : [
+      { type: "root", members: { owner: ["herb"] } },
+      { type : "rotate_key" },
+      {
+        type : "rotate_key_hidden"
+        corruptors :
+          sig3_corrupt_outer : (outer) ->
+            outer[3][1] ^= 0x01
```
Comment (heronhaye) [inline teamchains/inputs/hidden_bad_inner_hash.iced:17]: would be nice to comment these with the field name // InnerLinkID

File: teamchains/inputs/hidden_bad_parent.iced (hunk 4 above)
Comment (heronhaye) [reply teamchains/inputs/hidden_bad_parent.iced]: doesn't seem like right error

**Review by heronhaye — COMMENTED** [review]
Looks good except for the "unknown link type 1000" in hidden_bad_parent

I saw you're doing more, I came up with these for now

test for ignore if unsupported
test for bad signer kid
test for errors in checkLoadedRatchet
test for errors in checkNewLinkAgainstNewRatchets
test for errors in AddRatchet
test for errors in (r *RatchetBlind) check(tail sig3.Tail) (err error)

File: teamchains/inputs/hidden_bad_parent.iced (hunk 4 above)
Comment (maxtaco) [reply teamchains/inputs/hidden_bad_parent.iced]: It's skipped for now, since the test doesn't actually fail. There's a ticket for it.

**Comment by maxtaco** [comment]
Added your list to https://keybase.atlassian.net/browse/Y2K-434

File: teamchains/inputs/c.sh (hunk 1 above)
Comment (maxtaco) [reply teamchains/inputs/c.sh]: ok!

File: teamchains/inputs/hidden_bad_link_type.iced (hunk 3 above)
Comment (maxtaco) [reply teamchains/inputs/hidden_bad_link_type.iced]: fixed across the board

**Pull request merged**